src/modules/0.0.21/aws_cloudwatch/index.ts

```diff
@@ -53,36 +58,44 @@ class LogGroupMapper extends MapperBase<LogGroup> {
       return out;
     },
     read: async (ctx: Context, id?: string) => {
-      const client = (await ctx.getAwsClient()) as AWS;
+      const enabledRegions = (await ctx.getEnabledAwsRegions()) as string[];
       if (id) {
-        const rawLogGroup = await this.getLogGroup(client.cwClient, id);
-        if (!rawLogGroup) return;
-        return this.logGroupMapper(rawLogGroup);
+        for (const region of enabledRegions) {
+          const client = (await ctx.getAwsClient(region)) as AWS;
+          const rawLogGroup = await this.getLogGroup(client.cwClient, id);
+          if (!rawLogGroup) continue;
+          return this.logGroupMapper(rawLogGroup, region);
+        }
       } else {
-        const logGroups = (await this.getLogGroups(client.cwClient)) ?? [];
         const out = [];
-        for (const logGroup of logGroups) {
-          const outLog = this.logGroupMapper(logGroup);
-          if (outLog) out.push(outLog);
+        for (const region of enabledRegions) {
+          const client = (await ctx.getAwsClient(region)) as AWS;
+          const logGroups = (await this.getLogGroups(client.cwClient)) ?? [];
+          for (const i of logGroups) {
+            const lg = this.logGroupMapper(i, region);
+            if (lg) out.push(lg);
+          }
         }
         return out;
       }
     },
-    updateOrReplace: () => 'update',
+    updateOrReplace: (a: LogGroup, b: LogGroup) => (a.region !== b.region ? 'replace' : 'update'),
```

Comment:
This isn't necessary since it should never compare two objects with different regions as you have mentioned.